src/NServiceBus.Hosting.Windows/StartableAndStoppable/IWantToRunWhenEndpointStartsAndStops.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace NServiceBus
+{
+    using System.Threading.Tasks;
+    using JetBrains.Annotations;
+
+    /// <summary>
+    /// Implementers will be invoked when the endpoint starts up.
+    /// Dependency injection is provided for these types.
+    /// </summary>
+    [UsedImplicitly(ImplicitUseTargetFlags.WithMembers)]
+    public interface IWantToRunWhenEndpointStartsAndStops
+    {
+        /// <summary>
+        /// Method called at startup.
```

Comment:
In Bangkok we concluded that renaming the interface should be enough to guide the users since it is a new interface and we educate our customers what it means to return a task.